src/vehiclelist.cpp

```diff
@@ -116,15 +116,18 @@ bool GenerateVehicleSortList(VehicleList *list, const VehicleListIdentifier &vli
 
 	switch (vli.type) {
 		case VL_STATION_LIST:
-			for (const Vehicle *v : Vehicle::Iterate()) {
-				if (v->type == vli.vtype && v->IsPrimaryVehicle()) {
-					for (const Order *order : v->Orders()) {
-						if ((order->IsType(OT_GOTO_STATION) || order->IsType(OT_GOTO_WAYPOINT) || order->IsType(OT_IMPLICIT))
-								&& order->GetDestination() == vli.index) {
-							list->push_back(v);
-							break;
-						}
+			for (const OrderList *orderlist : OrderList::Iterate()) {
+				const Vehicle *v = orderlist->GetFirstSharedVehicle();
+				if (v->type != vli.vtype) continue;
+				/* Vehicle is a candidate, search for a matching order. */
```

Comment:
Bit of whitespace goes a long way here. Just a newline just before this comment makes already a visual break from what is going on, and makes it less blobby. Pretty sure with some well times newlines in other parts we increase readability :)